src/wallet/wallet.ts

```diff
@@ -479,8 +503,20 @@ class HathorWalletServiceWallet extends EventEmitter implements IHathorWallet {
     }
 
     if (!this.authToken || !validateJWTExpireDate(this.authToken)) {
-      const sign = this.signMessage(timestampNow);
+      let seed = this.seed;
+
+      if (!seed) {
+        // Request the client for the password
+        const password = usePassword ? usePassword : await this.requestPassword();
+
+        // Use it to get the words from the storage
+        seed = wallet.getWalletWords(password);
+      }
+
+      // @ts-ignore
```

Comment:
If we are confident `seed` will never be null here, maybe we should just do this instead:

```
      if (!seed) throw new Error('Invalid password');

      const sign = this.signMessage(seed, timestampNow);
```

Reply:
This was refactored on the purpose path PR, but made the change here on 8d4304d8db3adfa1ec64281d630ee3a9f48d49b2